src/cdk/collections/selection-model.ts

```diff
@@ -131,14 +131,7 @@ export class SelectionModel<T> {
    * Determines whether a value is selected.
    */
   isSelected(value: T): boolean {
-    if (this.compareWith) {
-      for (const otherValue of this._selection) {
-        if (this.compareWith(otherValue, value)) {
-          return true;
-        }
-      }
-      return false;
-    }
+    value = this._getConcreteValue(value);
```

Comment:
Maybe just inline this into the `has` call given that it isn't used anywhere else.

Reply:
Yes, edited !